storybookjs/storybook
# Addon-docs: Fix props table props sorting for PropTypes

Issue: https://github.com/storybookjs/storybook/issues/8539

## What I did

The sorting problems comes from react-docgen that mess up the original sorting order of the props by returning first the props that have a default value.

My fix consist of using the original definition of the PropTypes object to re-order the props based on their original definition order.

This only fix the problem for PropTypes. If other frameworks or type systems have the same problem, they will also need to be fixed.

## How to test

- Is this testable with Jest or Chromatic screenshots? Yes
- Does this need a new example in the kitchen sink apps? Yes
- Does this need an update to the documentation? No

Jest: `yarn jest --testPathPattern=handleProp`
cra-ts-kitchen-sink: ?path=/docs/propssort--page

![image](https://user-images.githubusercontent.com/794579/69023651-9438a380-098d-11ea-8776-16828252ab00.png)

If your answer is yes to any of these, please make sure to include it in your PR.
